atmo/atmo_test.go

```diff
@@ -0,0 +1,176 @@
+Strings.contains() --> built in go function.  ......... (14.29)
+
+3.
+/set/:key and /get/:key
+
+^^ to test: first request should have 200 response code and second has 200 and exact same value. /set/name and /get/name Botha re testing cache. (won't return any data for SET)- for check: ONLY checking that status code =200)
+pass variable into :key (like /set/name - Connor for example, and then /get as its the same to
+
+
+/set/:key *********      curl -d 'nyah' localhost:8080/set/name ***************.  Does not return anything- on a 200 response it doesn't return any day (this POSt request does not return data). Just check that the status code is 200.
+
+/get/:key *********      curl localhost:8080/get/name    *********** Should return name from above.
+
+4.
+/file/*file - "testing fetching a file from the static directory" ****** curl localhost:8080/file/main.md ********* will return contents of main.md *****
+RETURNS: [contents of main.md file] which is in this case: hello%
+
+Or: [other file example]
+curl localhost:8080/file/css/main.css
+RETURNS:
+.classname {
+	color: aliceblue;
+}%
+
+make a request to each file in Static directory (/css/main.css, /js/app/main.js, /main.md) and should return contents of the files.
+Compare return values of each return request to ensure it matches the contents of the file that you are asking for. Can jsut copy contents of file and read test or READ file (more automated).
+
+* = entire path (like /file/{css/main.css}), : = one element (vs /file/{main.css}) _ you can only have one element at the latter
+like: /file/css/main.css
+You can peek at each runnable that each fo these is using just to see how the runnable is writtena nd what APIs its using for general knowledge about atmo.
+
+/github: uses graphql function (LEAVE FOR NOW)- come back to because requires
+
+Ignore: /user--> (requires db) and /stream --> (requires web socket).
+
+*/
+//curl -d 'my friend' localhost:8080/hello
+//send post request to endpoint
+//should return "hello my friend"
+
+// 3 test important notes:
+// Method has to start w Test405Request
+// Param has to be (t *testing.T)
+// file name has to be _test.go
+
+// /hello       ****** curl -d 'my friend' localhost:8080/hello *********
+func TestEchoRequest(t *testing.T) {
+	atmo := atmoForBundle("../example-project/runnables.wasm.zip")
+
+	server, err := atmo.testServer()
+	if err != nil {
+		t.Error(err)
+		return
+	}
+
+	vt := vtest.New(server) //creating fake version of the server that we can send requests to and it will behave same was as if it was the real server
+
+	req, _ := http.NewRequest(http.MethodPost, "/hello", bytes.NewBuffer([]byte("my friend"))) //same way to use curl
+	// http.MethodGet.  file. bytes.NewBuffer([]byte("")//data from the body
+
+	resp := vt.Do(req, t) //execute the request and returns the response  (can use resp to do the checks)
+
+	resp.AssertStatus(200) //checks if response status code is 200 and if not it will fail
```

Comment:
We should also check to make sure the body is `hello my friend`

Reply:
handled below